modules/admin-ui/src/main/java/org/opencastproject/adminui/impl/ThumbnailImpl.java

```
@@ -0,0 +1,494 @@
+    // a URI.
+    final Attachment downscaled = composerService.convertImageSync(attachment, conversionProfile);
+    attachment.setURI(downscaled.getURI());
+  }
+
+  private URI updateExternalPublication(final MediaPackage mp, final MediaPackageElementFlavor trackFlavor)
+    throws IOException, NotFoundException, MediaPackageException, PublicationException,
+    EncoderException {
+    final Predicate<Attachment> flavorFilter = a -> a.getFlavor().matches(publishFlavor);
+    final Predicate<Attachment> tagsFilter = a -> Arrays.asList(a.getTags()).containsAll(publishTags);
+    final Predicate<Attachment> priorFilter = flavorFilter.and(tagsFilter);
+    final Tuple<URI, MediaPackageElement> result = updatePublication(mp, "api", priorFilter,
+      publishFlavor.applyTo(trackFlavor), publishTags, null);
+    if (result != null) {
+      return result.getA();
+    } else {
+      return null;
+    }
+  }
+
+  private InputStream tempInputStream() throws NotFoundException, IOException {
+    return workspace.read(tempThumbnail);
+  }
+
+  private void createTempThumbnail(final MediaPackage mp, final InputStream inputStream, final String contentType)
+    throws IOException {
+    tempThumbnailMimeType = MimeTypes.parseMimeType(contentType);
+    final String filename = "uploaded_thumbnail." + tempThumbnailMimeType.getSuffix().getOrElse("unknown");
+    final String originalThumbnailId = UUID.randomUUID().toString();
+    tempThumbnail = workspace.put(mp.getIdentifier().compact(), originalThumbnailId, filename, inputStream);
+    tempThumbnailFileName = "uploaded_thumbnail." + tempThumbnailMimeType.getSuffix().getOrElse("unknown");
+  }
+
+  private Optional<Publication> getPublication(final MediaPackage mp, final String channelId) {
+    return Arrays.stream(mp.getPublications()).filter(p -> p.getChannel().equalsIgnoreCase(channelId)).findAny();
+  }
+
+  private Publication replaceIgnoreExceptions(final MediaPackage mp, final String channelId,
```

Method name should be altered to match what it's doing (ie, it's not ignoring the exceptions, it's passing them upwards).

---

Actually, the method is not needed anymore. I removed it.